src/app/frontend/deploy/deployfromsettings.html

```diff
@@ -17,17 +17,19 @@
 <kd-help-section>
   <md-input-container class="md-block" md-is-error="ctrl.isNameError()">
     <label>App name</label>
-    <input ng-model="ctrl.name" name="name" namespace="ctrl.namespace" required kd-unique-name
-        ng-model-options="{ updateOn: 'default blur', debounce: { 'default': 500, 'blur': 0 } }">
-    <md-progress-linear class="kd-deploy-form-progress" md-mode="indeterminate"
-        ng-show="ctrl.form.name.$pending">
-    </md-progress-linear>
-    <ng-messages for="ctrl.form.name.$error" role="alert" multiple>
-      <ng-message when="required">Application name is required.</ng-message>
-      <ng-message when="uniqueName">
-        Application with this name already exists within namespace <i>{{ctrl.namespace}}</i>.
-      </ng-message>
-    </ng-messages>
+    <input ng-model="ctrl.name"
+           name="name"
+           namespace="ctrl.namespace"
+           required
+           kd-unique-name
+           ng-pattern="/^[a-z]+[a-z\-]*[a-z]*$/">
+
+    <div ng-messages="ctrl.form.name.$error" ng-if="ctrl.form.name.$touched">
```

Comment:
@bryk, no, unfortunately removing `md-is-error="ctrl.isNameError()"` altogether changes nothing to this behaviour. This behaviour is not visible on the 'Container Image' field as there are currently not enough error cases to be able to cause two consecutive errors, but on the 'Number of Pods' field, if you remove `ng-if="ctrl.form.replicas.$touched"`, and then enter a negative value, then click out to make the error message show, and then backspace to empty, you can see the same behaviour - the 'required' message does not appear. This 'Number of Pods' field has neither `...isNameError()` nor any asynchronous validation, so for the moment I am still inclined to think it is an AngularJS thing.
@batikanu , you are right, when you submit the form without touching the form.name, the line turns red, but no message appears. I can fix this though, if we are going to keep the `$touched` solution.